### PR TITLE
Simplify cylc version, dropping VERSION file and git info

### PR DESCRIPTION
**Pull request by kinow**
Fix for #2997 

- The `cylc.version` module has been deleted
- To minimize changes, instead of using `__version__`, I did a `from cylc import __version__ as CYLC_VERSION`
- Documentation updates were minimal, as most of the rest appears to be related to how Cylc is installed from `.tar.gz` or `git`. Both should be replaced by `setup.py` in #2990 later, where we will discuss ways to install Cylc (Conda, RPM, `pip`, etc)
- Fixed `make-tarball`, even though I realized it can be removed as part of #2990 

This is a blocker for #2991 as we need this to change how the `profile-battery` works too.

### CODE REVIEW

File: etc/dev-bin/make-tarball [hunk 1]
```diff
@@ -32,9 +37,9 @@ if [[ ! -d .git && ! -f .git ]]; then
 fi
 
 THIS_DIR=$( cd $( dirname $0 ); pwd; )
```
Comment (hjoliver) [inline etc/dev-bin/make-tarball]: This line not needed anymore.

Comment (kinow) [reply etc/dev-bin/make-tarball]: Line removed. Thanks!

File: lib/cylc/__init__.py [hunk 2]
```diff
@@ -64,3 +64,5 @@ def environ_path_add(dirs, key='PATH'):
 
 
 environ_init()
+
+__version__ = "8.0.0-a0"
```
[the remark this answers is not in the document]
Comment (kinow) [reply lib/cylc/__init__.py]: @hjoliver changed the tag in cylc/cylc, I saw the notification this morning and left in a tab in my mobile browser. But forgot to update this PR after that. Sorry. Change coming in a few seconds (Travis-CI busy tonight)

**Comment by kinow** [comment]
Version changed @matthewrmshin . Thanks!

**Pull request merged**